listenbrainz/webserver/static/js/src/personal-recommendations/PersonalRecommendationsModal.tsx

```diff
@@ -0,0 +1,263 @@
+    const input = event.target.value.replace(/\s\s+/g, " "); // remove line breaks and excessive spaces
+    if (input.length <= this.maxBlurbContentLength) {
+      this.setState({ blurbContent: input });
+    }
+  };
+
+  handleError = (error: string | Error, title?: string): void => {
+    const { newAlert } = this.props;
+    if (!error) {
+      return;
+    }
+    newAlert(
+      "danger",
+      title || "Error",
+      typeof error === "object" ? error.message : error
+    );
+  };
+
+  addUser = (user: string) => {
+    const { users } = this.state;
+    users!.push(user);
+    this.setState({ users: uniq(users), suggestions: [] });
+  };
+
+  removeUser = (user: string) => {
+    const { users } = this.state;
+    remove(users!, (element) => {
+      return element === user;
+    });
+    this.setState({ users });
+  };
+
+  searchUsers = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { followers } = this.state;
+    if (event.target.value) {
+      const suggestions = followers!.filter((username) => {
+        if (includes(username, event.target.value)) {
+          return username;
+        }
+        return null;
+      });
+      this.setState({ suggestions });
+    } else {
+      this.setState({ suggestions: [] });
+    }
+  };
+
+  submitPersonalRecommendation = async () => {
+    const { recordingToPersonallyRecommend, newAlert } = this.props;
+    const { blurbContent, users } = this.state;
+    const { APIService, currentUser } = this.context;
+
+    if (currentUser?.auth_token) {
+      const metadata: UserTrackPersonalRecommendationMetadata = {
+        artist_name: getArtistName(recordingToPersonallyRecommend),
+        track_name: getTrackName(recordingToPersonallyRecommend),
+        release_name: recordingToPersonallyRecommend!.track_metadata
+          .release_name,
+        recording_mbid: getRecordingMBID(recordingToPersonallyRecommend!),
+        recording_msid: getRecordingMSID(recordingToPersonallyRecommend!),
+        users,
+        blurb_content: blurbContent,
+      };
+      try {
+        const status = await APIService.submitPersonalRecommendation(
+          currentUser.auth_token,
+          currentUser.name,
+          metadata
+        );
+        if (status === 200) {
+          newAlert(
+            "success",
+            "You personally recommended a track!",
+            `${metadata.artist_name} - ${metadata.track_name}`
+          );
+          this.setState({ blurbContent: "" });
+        }
+      } catch (error) {
+        this.handleError(error, "Error while personally recommending");
+      }
+    }
+  };
+
+  render() {
+    const { recordingToPersonallyRecommend } = this.props;
+    if (!recordingToPersonallyRecommend) {
+      return null;
+    }
+    const { blurbContent, users, suggestions } = this.state;
+    const { track_name } = recordingToPersonallyRecommend.track_metadata;
+    const { artist_name } = recordingToPersonallyRecommend.track_metadata;
```

Comment:
These can be grouped like so:

```suggestion
    const { track_name, artist_name } = recordingToPersonallyRecommend.track_metadata;
```